crates/host/src/wasmbus/mod.rs

```diff
@@ -3651,6 +3802,66 @@ impl Host {
         Ok(ACCEPTED.into())
     }
 
+    #[instrument(skip(self, data))]
+    async fn handle_config_put(
+        &self,
+        entity_id: &str,
+        key: &str,
+        data: Bytes,
+    ) -> anyhow::Result<Bytes> {
+        debug!("handle config entry put");
+        // Simple check for a valid entity key. A valid public nkey is 56 chars and it should start
+        // with the proper character. It isn't worth the overhead of actually parsing the key, but
+        // if that is an issue in the future we can ensure it is a fully parsable nkey.
+        if entity_id.len() != 56 || (!entity_id.starts_with('M') && !entity_id.starts_with('V')) {
+            bail!("Invalid entity ID. The entity ID must be a valid public nkey for an actor or provider");
+        }
+        self.config_data
+            .put(format!("{entity_id}_{key}"), data)
+            .await
+            .context("Unable to store config data")?;
+        // We don't write it into the cached data and instead let the caching thread handle it as we
+        // won't need it immediately.
+
+        self.publish_event("config_set", event::config_set(entity_id, key))
+            .await?;
+
+        Ok(ACCEPTED.into())
+    }
+
+    #[instrument(skip(self))]
+    async fn handle_config_delete(&self, entity_id: &str, key: &str) -> anyhow::Result<Bytes> {
+        debug!("handle config entry deletion");
+
+        self.config_data
+            .purge(format!("{entity_id}_{key}"))
+            .await
+            .context("Unable to delete config data")?;
+
+        self.publish_event("config_deleted", event::config_deleted(entity_id, key))
+            .await?;
+
+        Ok(ACCEPTED.into())
+    }
+
+    #[instrument(skip(self))]
+    async fn handle_config_clear(&self, entity_id: &str) -> anyhow::Result<Bytes> {
+        debug!("handle config deletion");
+
+        let config_cache = self.config_data_cache.read().await;
```

Comment:
Why do we use `self.config_data` in the other functions but `self.config_data_cache` here?

Reply:
I was trying to shortcut a call to the database to list all keys, I can change this though

Reply:
Oh I remember now, I was trying to delete without listing all keys, so I would prefer not to change this if that's ok. Will add a comment though

Reply:
Oh, I see now, we're still using `self.config_data` inside `handle_config_delete` 👍